proto/legacy_amino/amino.proto

```diff
@@ -0,0 +1,69 @@
+syntax = "proto3";
+
+package legacy_amino;
+
+import "google/protobuf/descriptor.proto";
+
+// TODO(fdymylja): once we fully migrate to protov2 the go_package needs to be updated.
+// We need this right now because gogoproto codegen needs to import the extension.
+option go_package = "github.com/cosmos/cosmos-sdk/types/tx/legacy_amino";
+
+extend google.protobuf.MessageOptions {
+  // name is the string used when registering a concrete
+  // type into the Amino type registry, via the Amino codec's
+  // `RegisterConcrete()` method. This string MUST be at most 39
+  // characters long, or else the message will be rejected by the
+  // Ledger hardware device.
+  string name = 11110001;
+
+  // encoding describes the encoding format used by Amino
+  // for the message. The field type is chosen to be a string so that
+  // the value can either be:
+  // - a machine-readable string, such as "base64", "bech32" or "utf8",
+  // - or a human-readable string, for instance a short specification of how
+  // a big integer would be encoded using Amino.
+  //
+  // If left empty, then the Amino encoding is expected to be the same as the
+  // Protobuf one.
+  //
+  // Please note that `cosmos_proto.ScalarDescriptor` has an option with the
+  // same name `legacy_amino_encoding`. The cosmos_proto one however is used
+  // as a protobuf _field_ annotation, whereas this one operates on a protobuf
+  // _message_.
+  string encoding = 11110002;
```

Comment:
I'm still not sure how this would affect the encoding since isn't it the fields that matter when encoding messages?

Reply:
same question as https://github.com/cosmos/cosmos-sdk/pull/13501#discussion_r1006108527